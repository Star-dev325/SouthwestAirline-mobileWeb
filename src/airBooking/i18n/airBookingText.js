export default {
  AIR_BOOKING__AIRPORT_SELECTOR__DESCRIPTION: 'Select Airport',
  AIR_BOOKING__CHASE_PAYMENT_FAILURE__MESSAGE:
    'Please use a different form of payment to complete your booking. Your Southwest Rapid RewardsÂ® Credit Card will arrive in your mail shortly.',
  AIR_BOOKING__CHASE_PAYMENT_FAILURE__TITLE: 'Oops, something went wrong.',
  AIR_BOOKING__CORPORATE_BOOKING__ADD_IRN: 'Add internal reference number',
  AIR_BOOKING__CORPORATE_BOOKING__CANCEL: 'Cancel',
  AIR_BOOKING__CORPORATE_BOOKING__COMPANY_IRN: 'Company internal reference numbers',
  AIR_BOOKING__CORPORATE_BOOKING__DONE: 'Done',
  AIR_BOOKING__CORPORATE_BOOKING__EXPIRED_SUFFIX: ' - EXPIRED',
  AIR_BOOKING__CORPORATE_BOOKING__IRN: 'Internal reference number',
  AIR_BOOKING__CORPORATE_BOOKING__IRN_ABBREVIATED: 'Internal reference #',
  AIR_BOOKING__CORPORATE_BOOKING__LEARN_MORE: 'Learn more',
  AIR_BOOKING__CORPORATE_BOOKING__MANUAL_IRN_MESSAGE:
    'This internal reference number will not be saved to your account',
  AIR_BOOKING__CORPORATE_BOOKING__OPTIONAL_SUFFIX: '(optional)',
  AIR_BOOKING__CORPORATE_BOOKING__PAYMENT_INFORMATION_ON_FILE: 'Payment information on file',
  AIR_BOOKING__CORPORATE_BOOKING__SELECT: 'Select',
  AIR_BOOKING__CORPORATE_BOOKING__SELECT_COMPANY_TITLE: 'Select Company',
  AIR_BOOKING__CORPORATE_BOOKING__SELECT_CORPORATE_CARD: 'Select corporate card',
  AIR_BOOKING__CORPORATE_BOOKING__TAP_TO_SELECT_A_DIFFERENT_CARD: 'Tap to select a different corporate card',
  AIR_BOOKING__CORPORATE_BOOKING__TRAVELER_IRN: 'Personal internal reference numbers',
  AIR_BOOKING__ERROR__CID_NOT_AVAILABLE:
    'We were unable to process your request right now. Please try again or if you continue to have difficulties, please contact SWABIZ Help Desk (888-479-2249) for assistance.',
  AIR_BOOKING__ERROR__LFC__INVERTED_BOUNDS_DEPART: 'Departure date must be before return date',
  AIR_BOOKING__ERROR__LFC__INVERTED_BOUNDS_RETURN: 'Return date must be after departure date',
  AIR_BOOKING__ERROR__LFC__MISSING_BOUNDS: 'One or more required selections is missing, please try again.',
  AIR_BOOKING__FREQUENT_TRAVELER__ADD_NEW_BUTTON: 'Add a new frequent traveler',
  AIR_BOOKING__FREQUENT_TRAVELER__PAGE_HEADER: 'Frequent Travelers',
  AIR_BOOKING__LEARN_MORE: 'Learn More',
  AIR_BOOKING__LOW_FARE_CALENDAR: 'Low Fare Calendar',
  AIR_BOOKING_MULTI_SELECT_GROUP_UNAVAILABLE: 'Unavailable',
  AIR_BOOKING__PARENT_OR_GUARDIAN_FORM__CONTACT_INFORMATION_LABEL: 'CONTACT INFORMATION',
  AIR_BOOKING__PARENT_OR_GUARDIAN_FORM__CONTINUE: 'Continue',
  AIR_BOOKING__PARENT_OR_GUARDIAN_FORM__FLIGHT_INFORMATION_LABEL: 'FLIGHT INFORMATION',
  AIR_BOOKING__PARENT_OR_GUARDIAN_FORM__PARENT_OR_GUARDIAN_LABEL: 'PARENT OR GUARDIAN',
  AIR_BOOKING__PARTIAL_BOOKING__BUTTON_TEXT: 'Search Flights',
  AIR_BOOKING__PARTIAL_BOOKING__HEADER: 'Partial Confirmation Only',
  AIR_BOOKING__PARTIAL_BOOKING__MAIN_INSTRUCTION: 'Only part of your reservation was booked.',
  AIR_BOOKING__PARTIAL_BOOKING__NOT_CONFIRMED_HINT: "These passengers' reservations were not confirmed.",
  AIR_BOOKING__PARTIAL_BOOKING__SUB_INSTRUCTION: 'Please make your purchase for the below passengers separately.',
  AIR_BOOKING_PASSENGERS_ASSOCIATED_ADULT: 'ASSOCIATED ADULT',
  AIR_BOOKING__PASSENGERS_INTERNATIONAL_TRAVEL__INFORMATION: 'International travel information',
  AIR_BOOKING__PASSENGERS_INTERNATIONAL_TRAVEL__INFORMATION_TEXT:
    'You can wait to add passport and emergency contact until after booking. It will be required to check in.',
  AIR_BOOKING__PASSENGERS__ADULT_PERSONAL_INFO: 'PERSONAL INFO',
  AIR_BOOKING__PASSENGERS__COMPLETE: '(Complete)',
  AIR_BOOKING__PASSENGERS__DATE_OF_BIRTH: 'DATE OF BIRTH',
  AIR_BOOKING__PASSENGERS__EMAIL_RECEIPT_TO: 'EMAIL RECEIPT TO',
  AIR_BOOKING__PASSENGERS__GENDER: 'GENDER',
  AIR_BOOKING__PASSENGERS__LAPCHILD_INFO:
    'A Child over 14 days old and under 2 years of age on the date of travel may be carried on the lap of an accompanying Passenger 12 years of age or older.',
  AIR_BOOKING__PASSENGERS__LAPCHILD_PROOF_INFO:
    'A birth certificate or other government-issued identification bearing the birth date of the Child is required upon request.',
  AIR_BOOKING__PASSENGERS__LAPCHILD_PROOF_TITLE: 'Proof of age identification requirements',
  AIR_BOOKING__PASSENGERS__LOGIN_BANNER_TITLE: 'Log in for faster checkout',
  AIR_BOOKING__PASSENGERS__NAME_HELP_INFO:
    'First, middle, and last name must match government-issued photo identification.',
  AIR_BOOKING__PASSENGERS__OPTIONAL: '(Optional)',
  AIR_BOOKING__PASSENGERS__PARTIAL_BOOKING_PASSENGER_TITLE: 'PASSENGER',
  AIR_BOOKING__PASSENGERS__RAPID_REWARDS_NUMBER: 'RAPID REWARDS / ACCT#',
  AIR_BOOKING__PASSENGERS__TITLE: 'Passenger',
  AIR_BOOKING__PASSENGER_PERSONAL_INFO_FORM__FREQUENT_TRAVELER_BUTTON: 'Choose a frequent traveler',
  AIR_BOOKING__PASSENGER_PERSONAL_INFO_FORM__FREQUENT_TRAVELER_EDIT_INSTRUCTION:
    'You may update this passenger but, the changes will not be updated in your account.',
  AIR_BOOKING__PASSENGER_PERSONAL_INFO_FORM__SAVE_FREQUENT_TRAVELER:
    'Save frequent traveler to my account with purchase',
  AIR_BOOKING__PURCHASE_CONFIRMATION_MESSAGES__HEADER: 'Confirmation',
  AIR_BOOKING__SEARCH_FLIGHTS__FIND_FLIGHTS: 'Find flights',
  AIR_BOOKING__SEARCH_FLIGHTS__ONE_WAY: 'ONE WAY',
  AIR_BOOKING__SEARCH_FLIGHTS__PROMO_CODE: 'Promo Code (optional)',
  AIR_BOOKING__SEARCH_FLIGHTS__RECENT: 'Recent',
  AIR_BOOKING__SEARCH_FLIGHTS__ROUND_TRIP: 'ROUND TRIP',
  AIR_BOOKING__SEARCH_FLIGHTS__SHOW_FARES_IN: 'Show fares in:',
  AIR_BOOKING__SEARCH_FLIGHTS__TITLE: 'Book a Flight',
  AIR_BOOKING__SHOPPING__NEXT_DAY: 'Next Day',
  AIR_BOOKING__SHOPPING__OVERNIGHT: 'Overnight',
  AIR_BOOKING__SUMMARY_PAYMENT__KIND_REMINDER_MESSAGE:
    "You'll be able to review this trip before completing your purchase.",
  AIR_BOOKING__SUMMARY_PAYMENT__SHORT_CARD_NUMBER_MESSAGE: 'Last 4 digits: ',
  AIR_BOOKING__SUMMARY__CONTINUE: 'Continue',
  AIR_BOOKING__YOUNG_TRAVELER_EDIT__PAGE_TITLE: 'Parent or Guardian',
  AIR_BOOKING__YOUNG_TRAVELER__PAGE_TITLE: 'Young Traveler'
};
