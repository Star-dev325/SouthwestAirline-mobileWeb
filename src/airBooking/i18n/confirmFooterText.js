export default {
  AIR_BOOKING__NEED_A: 'Need a',
  AIR_BOOKING__RENTAL_CAR: 'rental car?',
  AIR_BOOKING__BOOK_IT_NOW: 'Book it now'
};
