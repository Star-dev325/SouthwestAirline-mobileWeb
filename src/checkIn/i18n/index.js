import addtionalPassportInfoFormText from 'src/checkIn/i18n/additionalPassportInfoFormText';
import checkInErrorsText from 'src/checkIn/i18n/checkInErrorsText';
import checkInText from 'src/checkIn/i18n/checkInText';
import chooseMobileBoardingPassesText from 'src/checkIn/i18n/chooseMobileBoardingPassesText';
import hazmatText from 'src/checkIn/i18n/hazmatText';
import mobileBoardingPassText from 'src/checkIn/i18n/mobileBoardingPassText';
import permanentResidentCardText from 'src/checkIn/i18n/permanentResidentCardText';
import trackCheckBagsText from 'src/checkIn/i18n/trackCheckBagsText';
import visaText from 'src/checkIn/i18n/visaText';

const checkIn = {
  ...addtionalPassportInfoFormText,
  ...checkInErrorsText,
  ...checkInText,
  ...chooseMobileBoardingPassesText,
  ...hazmatText,
  ...mobileBoardingPassText,
  ...permanentResidentCardText,
  ...trackCheckBagsText,
  ...visaText
};

export default checkIn;
