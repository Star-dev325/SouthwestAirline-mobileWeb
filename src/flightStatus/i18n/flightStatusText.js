export default {
  FLIGHT_STATUS__NOW_BOARDING: 'Now Boarding'
};
