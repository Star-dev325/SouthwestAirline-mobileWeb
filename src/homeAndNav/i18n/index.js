import emailEnrollText from 'src/homeAndNav/i18n/emailEnrollText';
import homeNavText from 'src/homeAndNav/i18n/homeNavText';
import inFlightText from 'src/homeAndNav/i18n/inFlightText';
import menuListText from 'src/homeAndNav/i18n/menuListText';
import navDrawerText from 'src/homeAndNav/i18n/navDrawerText';
import offersText from 'src/homeAndNav/i18n/offersText';

const homeAndNavText = {
  ...emailEnrollText,
  ...homeNavText,
  ...inFlightText,
  ...menuListText,
  ...navDrawerText,
  ...offersText
};

export default homeAndNavText;
