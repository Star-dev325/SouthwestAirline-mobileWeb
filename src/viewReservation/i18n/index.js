import viewReservationDetailText from 'src/viewReservation/i18n/viewReservationDetailText';
import boardingInfoText from 'src/viewReservation/i18n/boardingInfoText';
import dayOfTravelText from 'src/viewReservation/i18n/dayOfTravelText';
import viewReservationText from 'src/viewReservation/i18n/viewReservationText';

const viewReservation = {
  ...boardingInfoText,
  ...dayOfTravelText,
  ...viewReservationText,
  ...viewReservationDetailText
};

export default viewReservation;
