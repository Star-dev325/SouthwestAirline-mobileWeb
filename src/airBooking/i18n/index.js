import airBookingText from 'src/airBooking/i18n/airBookingText';
import confirmFooterText from 'src/airBooking/i18n/confirmFooterText';
import lowFareCalendarText from 'src/airBooking/i18n/lowFareCalendarText';
import popupText from 'src/airBooking/i18n/popupText';
import splitPayText from 'src/airBooking/i18n/splitPayText';
import confirmPassportRequiredText from 'src/airBooking/i18n/confirmPassportRequiredText';

const airBooking = {
  ...airBookingText,
  ...confirmFooterText,
  ...confirmPassportRequiredText,
  ...lowFareCalendarText,
  ...popupText,
  ...splitPayText
};

export default airBooking;
