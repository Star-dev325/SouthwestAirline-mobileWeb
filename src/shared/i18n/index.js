import lapChildText from 'src/airBooking/i18n/lapChildText';
import airportListText from 'src/shared/i18n/airportListText';
import airportSelectorText from 'src/shared/i18n/airportSelectorText';
import billingInfoText from 'src/shared/i18n/billingInfoText';
import boardingInformationText from 'src/shared/i18n/boardingInformationText';
import boundSelectText from 'src/shared/i18n/boundSelectText';
import breakdownToggleNamesText from 'src/shared/i18n/breakdownToggleNamesText';
import buttonText from 'src/shared/i18n/buttonText';
import calendarText from 'src/shared/i18n/calendarText';
import carReservationText from 'src/shared/i18n/carReservationText';
import commonSharedText from 'src/shared/i18n/commonSharedText.js';
import contactMethodOptionsText from 'src/shared/i18n/contactMethodOptionsText';
import contactMethodText from 'src/shared/i18n/contactMethodText';
import earlyBirdInPathText from 'src/shared/i18n/earlyBirdInPathText';
import earlyBirdText from 'src/shared/i18n/earlyBirdText';
import errorPageText from 'src/shared/i18n/errorPageText';
import errorText from 'src/shared/i18n/errorText';
import flightProductText from 'src/shared/i18n/flightProductText';
import flightSegmentDetailsText from 'src/shared/i18n/flightSegmentDetailsText';
import flightsSummaryHeaderText from 'src/shared/i18n/flightsSummaryHeaderText';
import flightStatusText from 'src/shared/i18n/flightStatusText';
import footerText from 'src/shared/i18n/footerText';
import fundResultItemText from 'src/shared/i18n/fundResultItemText.js';
import hazmatDisclaimerText from 'src/shared/i18n/hazmatDisclaimerText';
import headerButtonText from 'src/shared/i18n/headerButtonText';
import labelStringText from 'src/shared/i18n/labelStringText';
import navGroupItemText from 'src/shared/i18n/navGroupItemText';
import passengerReservationTitleText from 'src/shared/i18n/passengerReservationTitleText';
import passportPageText from 'src/shared/i18n/passportPageText';
import paymentLinksText from 'src/shared/i18n/paymentLinksText';
import placeholderText from 'src/shared/i18n/placeholderText';
import priceLineTitlesText from 'src/shared/i18n/priceLineTitlesText';
import promoCodesText from 'src/shared/i18n/promoCodesText';
import purchaseSummaryFormText from 'src/shared/i18n/purchaseSummaryFormText';
import recentSearchesText from 'src/shared/i18n/recentSearchesText';
import redressAndKnownTravelerFieldsText from 'src/shared/i18n/redressAndKnownTravelerFieldsText';
import refundMethodsText from 'src/shared/i18n/refundMethodsText';
import repricingText from 'src/shared/i18n/repricingText';
import reservationPassengerText from 'src/shared/i18n/reservationPassengerText';
import reviewFooterText from 'src/shared/i18n/reviewFooterText';
import savedCreditCardsTitleText from 'src/shared/i18n/savedCreditCardsTitleText';
import selectPassengersPageText from 'src/shared/i18n/selectPassengersPageText';
import shareItineraryText from 'src/shared/i18n/shareItineraryText';
import specialAssistanceText from 'src/shared/i18n/specialAssistanceText';
import spinnerText from 'src/shared/i18n/spinnerText';
import summaryInsufficientText from 'src/shared/i18n/summaryInsufficientText';
import tripBookedText from 'src/shared/i18n/tripBookedText';

const shared = {
  ...airportListText,
  ...airportSelectorText,
  ...billingInfoText,
  ...boardingInformationText,
  ...boundSelectText,
  ...breakdownToggleNamesText,
  ...buttonText,
  ...calendarText,
  ...carReservationText,
  ...commonSharedText,
  ...contactMethodOptionsText,
  ...contactMethodText,
  ...earlyBirdInPathText,
  ...earlyBirdText,
  ...errorPageText,
  ...errorText,
  ...flightProductText,
  ...flightSegmentDetailsText,
  ...flightsSummaryHeaderText,
  ...flightStatusText,
  ...footerText,
  ...fundResultItemText,
  ...hazmatDisclaimerText,
  ...headerButtonText,
  ...labelStringText,
  ...lapChildText,
  ...navGroupItemText,
  ...passengerReservationTitleText,
  ...passportPageText,
  ...paymentLinksText,
  ...placeholderText,
  ...priceLineTitlesText,
  ...promoCodesText,
  ...purchaseSummaryFormText,
  ...recentSearchesText,
  ...redressAndKnownTravelerFieldsText,
  ...refundMethodsText,
  ...repricingText,
  ...reservationPassengerText,
  ...reviewFooterText,
  ...savedCreditCardsTitleText,
  ...selectPassengersPageText,
  ...shareItineraryText,
  ...specialAssistanceText,
  ...spinnerText,
  ...summaryInsufficientText,
  ...tripBookedText
};

export default shared;
