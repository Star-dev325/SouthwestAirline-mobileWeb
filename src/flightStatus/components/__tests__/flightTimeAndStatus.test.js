import i18n from '@swa-ui/locale';
import { render } from '@testing-library/react';
import React from 'react';
import FlightTimeAndStatus from 'src/flightStatus/components/flightTimeAndStatus';

describe('flightTimeAndStatus', () => {
  const defaultProps = {
    flightStatus: 'ON TIME',
    statusType: 'POSITIVE',
    timeString: '2015-04-28T19:15:00',
    isNextDay: false,
    statusIconPosition: 'UPPER'
  };

  describe('when initialize', () => {
    it('should render properly', () => {
      const { container } = createComponent(defaultProps);

      expect(container).toMatchSnapshot();
    });

    it('should render with correct class name', () => {
      const { container } = createComponent(defaultProps);

      expect(container.querySelector('.flight-time-status')).not.toBeNull();
      expect(container.querySelector('.flight-time-status--on-time')).not.toBeNull();
    });
  });

  describe('when flightStatus is given', () => {
    it('should render correctly when the flight is delayed', () => {
      const { container } = createComponent({
        flightStatus: 'DELAYED',
        statusType: 'NEGATIVE',
        timeString: '2015-04-28T19:15:00',
        isNextDay: false,
        statusIconPosition: 'UNDER',
        originalTime: '19:00'
      });

      expect(container.querySelector('.segment-status-col')).toMatchSnapshot();
    });
  });

  describe('when flightStatus is not given', () => {
    it('should render FlightTime but no FlightStatusIcon', () => {
      const { container } = createComponent({
        timeString: '2015-04-28T19:15:00',
        isNextDay: false
      });

      const flightTime = container.querySelector('.flight-time--time');
      const flightStatusIcon = container.querySelector('.segment-status-col');

      expect(flightTime).not.toBeNull();
      expect(flightStatusIcon).toBeNull();
    });
  });

  describe('isNextDay & isOvernight', () => {
    it('should render FlightTime with nextDay icon when isNextDay is true', () => {
      const { container } = createComponent({
        timeString: '2015-04-28T19:15:00',
        isNextDay: true
      });

      expect(container.querySelector('.flight-time-status').textContent).toContain(`7:15PM${i18n('AIR_BOOKING__SHOPPING__NEXT_DAY')}`);
    });

    it('should render FlightTime without children when isNextDay is false', () => {
      const { container } = createComponent({
        timeString: '2015-04-28T19:15:00',
        isNextDay: false
      });

      expect(container.querySelector('[da-qa="next-day-indicator"]')).toBeNull();
    });

    it('should render FlightTime with Overnight icon when isOvernight is true', () => {
      const { container } = createComponent({
        timeString: '2015-04-28T19:15:00',
        isOvernight: true
      });

      expect(container.querySelector('.flight-time-status').textContent).toContain(`7:15PM${i18n('AIR_BOOKING__SHOPPING__OVERNIGHT')}`);
    });

    it('should render FlightTime without children when isOvernight is false', () => {
      const { container } = createComponent({
        timeString: '2015-04-28T19:15:00',
        isOvernight: false
      });

      expect(container.querySelector('[da-qa="overnight-indicator"]')).toBeNull();
    });

    it('should render FlightTime with only Overnight icon when isOvernight and isNextDay are both true', () => {
      const { container } = createComponent({
        timeString: '2015-04-28T19:15:00',
        isOvernight: true,
        isNextDay: true
      });

      expect(container.querySelector('.flight-time-status').textContent).toContain(`7:15PM${i18n('AIR_BOOKING__SHOPPING__OVERNIGHT')}`);
    });
  });

  const createComponent = (props) => render(<FlightTimeAndStatus {...props} />);
});
