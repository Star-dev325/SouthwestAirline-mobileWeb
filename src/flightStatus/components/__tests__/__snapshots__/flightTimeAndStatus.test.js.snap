// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`flightTimeAndStatus when flightStatus is given should render correctly when the flight is delayed 1`] = `
<span
  class="segment-status-col flight-status negative"
>
  <i
    class="icon icon_home-flight-status"
  />
  <span
    class="flight-status--text"
  >
    DELAYED
  </span>
  <p
    class="flight-status--sub-text"
  >
     
    (was 
    7:00
    <span
      class="time-period--subtext"
    >
      PM
    </span>
    )
  </p>
</span>
`;

exports[`flightTimeAndStatus when initialize should render properly 1`] = `
<div>
  <div
    class="flight-time-status flight-time-status--on-time"
  >
    <div>
      <span
        class="segment-status-col flight-status positive"
      >
        <i
          class="icon icon_home-flight-status"
        />
        <span
          class="flight-status--text"
        >
          ON TIME
        </span>
      </span>
    </div>
    <div
      class="flight-time"
    >
      <div
        class="flight-time--time"
      >
        <span>
          7:15
        </span>
        <span
          class="time-period"
        >
          PM
        </span>
      </div>
    </div>
  </div>
</div>
`;
